### PR TITLE
Extend support record with new fields

### PR DESCRIPTION
**Pull request by aunroel**
Refactors existing `SupportRecord` model with new data per 'Support Cas has [SupportLogEntry]' relationship

### CODE REVIEW

File: data_model.mdd [hunk 1]
```diff
@@ -355,13 +358,27 @@
         bool enable_notification "Parental leave or career breaks require notifications if they are within 30days"
     }
     SupportRecord {
-        int id
+        int id PK
         int trainee_id FK
-        date date "Support Record Date Entered onto DB, Null: True"
+        int case_type_id FK "Support Issue Type, Null: True"
+        date open_date "Support Record Date Entered onto DB, Null: True"
```
Comment (William-Blackie) [inline data_model.mdd:364]: The open date shouldn't be null=True, as this is a new record we don't need to worry about inconsistent data here, it should just be the current date the time of save.

File: data_model.mdd [hunk 2]
```diff
@@ -355,13 +358,27 @@
...
         string(MAX) comments "Support Description, Null: True"
+        int support_status_id FK "Support Status, Null: True"
```
Comment (William-Blackie) [inline data_model.mdd:366]: Note: perhaps we should also enforce this?

**Review by William-Blackie — APPROVED** [review]
Hey, this looks good @aunroel, just a couple of nitpicks

File: data_model.mdd (hunk 1 above)
Comment (aunroel) [reply data_model.mdd:364]: ah, great idea re current date, nullability was mainly defensive against incomplete data imports

plus I am enforcing open date on the UI at record creation. will remove nullability

File: data_model.mdd (hunk 2 above)
Comment (aunroel) [reply data_model.mdd:366]: So this is "fun"

Since I am extending existing `SupportRecord` - import Excel data won't have such a column

my idea was to leave it nullable for import, after which a dev would go and manually add respective status to the db (via OutSystems) after consulting with school 

now that I think of it - we can just default all imported records to 'Closed' status, which can be edited by users later on

TLDR - removing nullability
